clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -88,7 +88,10 @@
             new ApiException("Some of the committing partitions are not assigned the committer")),
     INVALID_COMMIT_OFFSET_SIZE(28,
             new ApiException("The committing offset data size is not valid")),
-    AUTHORIZATION_FAILED(29, new ApiException("Request is not authorized."));
+    AUTHORIZATION_FAILED(29,
+            new ApiException("Request is not authorized.")),
+    UNSUPPORTED_VERSION(30,
+            new UnsupportedVersionException("The version is not supported."));
```

Comment:
The version check actually is not only for request but also for response and magic byte. So the error message might need to be general.